### ISSUE TITLE
setting body to `z.never()` on `POST`, `PUT`, `PATCH`, `DELETE` contracts will cause an unknown error on the server

### ISSUE THREAD

**Issue by awilderink** [ts-rest/ts-rest#307]
Assigning a `z.never()` or `z.undefined()` on the `body` to `POST`, `PUT`, `PATCH` or `DELETE` methods will make `body` on the client side not required to be defined, but when passing the request to the server, it will fail with an unknown error. 

**Workaround**
  * Set `body` to `z.any()` or `z.unknown()`
  * Pass an empty object to `body` in the client request
  
Is it really necessary to always have the body defined in non `GET` requests? Since this is also not required by the REST specification

**Comment by schowdhuri** [comment]
This tripped me up too. I vote `body` be made optional

**Comment by vaartaa** [comment]
I'm experiencing the same issue. While the workaround of adding an empty body does work, it doesn't seem like a suitable solution and appears to introduce unnecessary overhead. **It would be better if the body could be made optional instead.**

**Comment by michaelangelo-io** [comment]
We'll look into this. Not sure whether to use enhancement or bug label

**Comment by firxworx** [comment]
See my issue #383 for a related requirement/issue for when body is set to optional: the generated ts-rest client must also _not_ send application/json content-type header with the request when there is no body.

Otherwise a rigorous server framework like Fastify (and presumably NestJS with Fastify option) will have an issue and throw an error.

Since #383 impacts standard CRUD functionality and implementation of a typical REST-ish JSON API using Fastify I think a "bug" label and prioritization is warranted vs. "enhancement".

**Comment by Gabrola** [comment]
Resolved in #547

**Comment by russriser** [comment]
I think this issue was re-introduced at some point. I haven't been able to avoid sending a body with DELETE requests regardless of what type I assign to the request body in the contract. 

Seems like #772 and #69

Error: 
```
SyntaxError: Unexpected end of JSON input
    at JSON.parse (<anonymous>)
    at parseJSONFromBytes (node:internal/deps/undici/undici:5738:19)
    at successSteps (node:internal/deps/undici/undici:5719:27)
    at consumeBody (node:internal/deps/undici/undici:5725:9)
    at TsRestRequest.json (node:internal/deps/undici/undici:5665:18)
    at evaluateContent (/var/task/index.js:42323:44)
    at Object.fetch (/var/task/index.js:42085:60)
```

UPDATE: I just had 'content-type': 'application/json' set in my client configuration's baseHeaders. Removing that fixed the error and still allows other requests to complete as expected.
